wdl-analysis/src/handlers/semantic_tokens.rs

```diff
@@ -0,0 +1,312 @@
+            Some(SemanticTokenType::NUMBER)
+        }
+        k if k.is_keyword() => Some(SemanticTokenType::KEYWORD),
+        k if is_operator(k) => Some(SemanticTokenType::OPERATOR),
+        SyntaxKind::BooleanTypeKeyword
+        | SyntaxKind::IntTypeKeyword
+        | SyntaxKind::FloatTypeKeyword
+        | SyntaxKind::StringTypeKeyword
+        | SyntaxKind::FileTypeKeyword
+        | SyntaxKind::DirectoryTypeKeyword
+        | SyntaxKind::ArrayTypeKeyword
+        | SyntaxKind::PairTypeKeyword
+        | SyntaxKind::MapTypeKeyword
+        | SyntaxKind::ObjectTypeKeyword => Some(SemanticTokenType::TYPE),
+
+        SyntaxKind::Ident => resolve_identifier_ty(token, &parent, document),
+        _ => None,
+    };
+
+    // TODO: SemanticTokenModifiers
+    // Token types are looked up by index, so a tokenType value of 1 means
+    // tokenTypes[1]. Since a token type can have n modifiers, multiple token
+    // modifiers can be set by using bit flags, so a tokenModifier value of 3 is
+    // first viewed as binary 0b00000011, which means [tokenModifiers[0],
+    // tokenModifiers[1]] because bits 0 and 1 are set.
+    //
+    // https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#semanticTokenModifiers
+    ty.map(|t| (t, 0))
+}
+
+/// Checks if a [`SyntaxKind`] corresponds to an operator.
+fn is_operator(kind: SyntaxKind) -> bool {
+    matches!(
+        kind,
+        SyntaxKind::Plus
+            | SyntaxKind::Minus
+            | SyntaxKind::Slash
+            | SyntaxKind::Percent
+            | SyntaxKind::Asterisk
+            | SyntaxKind::Exponentiation
+            | SyntaxKind::Equal
+            | SyntaxKind::NotEqual
+            | SyntaxKind::Less
+            | SyntaxKind::LessEqual
+            | SyntaxKind::Greater
+            | SyntaxKind::GreaterEqual
+            | SyntaxKind::LogicalAnd
+            | SyntaxKind::LogicalOr
+            | SyntaxKind::Exclamation
+            | SyntaxKind::Assignment
+            | SyntaxKind::QuestionMark
+            | SyntaxKind::Dot
+            | SyntaxKind::Colon
+    )
+}
+
+/// Resolves the semantic type of an identifier token based on its context.
+///
+/// This inspects the identifier's parent and ancestor nodes in the CST to
+/// determine its role, checking for:
+/// 1. Definition sites (task, workflow, struct, variable/parameter
+///    declarations).
+/// 2. Type references.
+/// 3. Function or task/workflow calls.
+/// 4. Import namespace aliases.
+/// 5. Member access expressions.
+/// 6. If none of the above apply, it falls back to looking up the identifier in
+///    the current scope to determine its type.
+fn resolve_identifier_ty(
+    token: &SyntaxToken,
+    parent: &SyntaxNode,
+    document: &Document,
+) -> Option<SemanticTokenType> {
+    if let Some(t) = TaskDefinition::cast(parent.clone())
+        && t.name().inner() == token
+    {
+        return Some(SemanticTokenType::MACRO);
```

Comment:
I wonder if this should just be `FUNCTION` to be keep it simple (I think of a task as more of a function than a macro).

Reply:
There's `FUNCTION` and `METHOD`. Which one would you prefer 🤔